src/cli/export/onedrive.go

```diff
@@ -0,0 +1,215 @@
+package export
+
+import (
+	"context"
+	"io"
+	"os"
+	ospath "path"
+
+	"github.com/alcionai/clues"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/alcionai/corso/src/cli/flags"
+	. "github.com/alcionai/corso/src/cli/print"
+	"github.com/alcionai/corso/src/cli/repo"
+	"github.com/alcionai/corso/src/cli/utils"
+	"github.com/alcionai/corso/src/internal/common/dttm"
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/internal/observe"
+	"github.com/alcionai/corso/src/pkg/control"
+	"github.com/alcionai/corso/src/pkg/export"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+// called by export.go to map subcommands to provider-specific handling.
+func addOneDriveCommands(cmd *cobra.Command) *cobra.Command {
+	var (
+		c  *cobra.Command
+		fs *pflag.FlagSet
+	)
+
+	switch cmd.Use {
+	case exportCommand:
+		c, fs = utils.AddCommand(cmd, oneDriveExportCmd())
+
+		c.Use = c.Use + " " + oneDriveServiceCommandUseSuffix
+
+		// Flags addition ordering should follow the order we want them to appear in help and docs:
+		// More generic (ex: --user) and more frequently used flags take precedence.
+		fs.SortFlags = false
+
+		flags.AddBackupIDFlag(c, true)
+		flags.AddOneDriveDetailsAndRestoreFlags(c)
+		flags.AddExportConfigFlags(c)
+		flags.AddFailFastFlag(c)
+		flags.AddCorsoPassphaseFlags(c)
+		flags.AddAWSCredsFlags(c)
+	}
+
+	return c
+}
+
+const (
+	oneDriveServiceCommand          = "onedrive"
+	oneDriveServiceCommandUseSuffix = "--backup <backupId> <destination>"
+
+	//nolint:lll
+	oneDriveServiceCommandExportExamples = `# Export file with ID 98765abcdef in Bob's last backup (1234abcd...) to my-exports directory
+corso export onedrive --backup 1234abcd-12ab-cd34-56de-1234abcd --file 98765abcdef my-exports
```

Comment:
Coming back to this after my prior approval, because I missed it on the first review:

`--file 98765abcdef my-exports` doesn't look like a compilable example.  Is this supposed to be equivalent to `corso export onedrive my-destination --backup . --file .`?  If so, I'm not comfortable with the design of injecting parameters _behind_ the flags.  That's non-standard for our cli, and is (imo) a bit confusing in layout and readability.

At the very least, I'd like to be sure the cli functions as shown in the suggestion below.  Then, we should have all examples follow that pattern.  If it's possible for the user to put the destination after the flags, and they choose to do that on their own, that's another matter.  But for our docs, lets keep it consistent with commands and command arguments before flags.

```suggestion
corso export onedrive my-exports --backup 1234abcd-12ab-cd34-56de-1234abcd --file 98765abcdef
```

Reply:
Ahh, both options should work. Let me change the example to have he destination to be mentioned before passing in any flags.

Reply:
Much appreciated.  Once this PR is rebased I'll give it another look over and approval.